### ISSUE TITLE
DB setup script not executing, project fails to load

### ISSUE THREAD

**Issue by FracturedSarcasm** [AdKats/BFACP#298]
We've been running the PRoCon Chat, GUID, Stats and Map Logger v1.0.0.3 along with Adkats v7.6.0.5 successfully for quite some time. I've downloaded the latest stable version of BFACP (v2.0.3) and followed the directions in the read me file. I've created a subdomain admin.btxgamers.com and made the document root admin.btxgamers.com/public. The .env.php file has all of the appropriate info as needed but when I load admin.btxgamers.com I get an error message.

We are running MySQL: v5.7.27 and PHP v7.2.24.

Thoughts, ideas, suggestions as to why it's not working?

@Prophet731 - I appreciate all of the hardwork put into this application.
[
![BFACP Error](https://user-images.githubusercontent.com/49985936/79678137-fd2a7e80-81ac-11ea-8948-dd27fde8c66e.PNG)
](url)

**Comment by Prophet731** [comment]
The issue is php7. It doesn't work with it with the latest version.

**Comment by toddpiper79** [comment]
I downgraded the php version to 5.6 (also tried 5.5 to match the readme file) and I'm getting the following error:

![image](https://user-images.githubusercontent.com/49985936/79700767-0703be80-824d-11ea-8683-7a8389c04691.png)

I opened the URL to generate a new 32 length key without symbols and updated the .env.php file's secret key section. Error still persists.

**Comment by Prophet731** [comment]
Looks like it's failing at line 32. What's the full error say? It says at the top of that code block.

**Comment by FracturedSarcasm** [comment]
Here's the entire msg:

![image](https://user-images.githubusercontent.com/49985936/79705392-67552900-826a-11ea-9753-9f6a05a3f286.png)

**Comment by Prophet731** [comment]
Is the entire `vendor` folder uploaded? That's saying it's missing files. Try reuploading that folder.

**Comment by FracturedSarcasm** [comment]
Yeah I had verified all directories were uploaded before launching the app. I'll re-upload the Vendor directory again. Does the v.2.0.3 have all the files or should I try a different version?

**Comment by FracturedSarcasm** [comment]
Decided to scrap the entire project and upload the entire thing using a different FTP client (Filezilla can suck it). The e error I'm getting now:

![image](https://user-images.githubusercontent.com/49985936/79774380-9493ec80-82e7-11ea-86d5-e3aa342a600a.png)

It looks like the DB setup script isn't firing; when I check my DB I don't see any bfacp_* tables. In a related ticket, someone had mentioned using:

php artisan migrate --force
php artisan db:seed --force

I just don't know where to use those commands so I started digging around repo again and found a setup script that looks like it creates all of the necessary tables. Should I run that or...?

Again, I appreciate the feedback dude. It's been a big help.

**Comment by cmoore-mayesh** [comment]
> php artisan migrate --force
> php artisan db:seed --force

I'm at the same spot here -- I agree its not triggering the initial setup of the database tables.  When I run 'php artisan migrate --force' I get an 'Error in exception handler.' and a return to the command prompt.

**Comment by Prophet731** [comment]
@cmoore-mayesh 

Make sure there isn't a database connection issue (right credentials, hostname, etc). 

Also, I would recommend that you install the latest de version I did. 

https://github.com/AdKats/BFACP/tree/dev-laravel5#installing-dev-version

**Comment by FracturedSarcasm** [comment]
I followed the directions to install the dev version and the first part worked like a charm. The files were pulled down, the permissions were the folder were modified and I was able to modify the .env file.

I copied and pasted each 'php artisan ...' line to eliminate the possibility of a typo. Nothing happened. When I hit the enter button, the terminal just returned me to the prompt. When I refreshed the DB no tables were created. So naturally going to the BFACP site did not work.

**Comment by Prophet731** [comment]
@FracturedSarcasm What shows up when you run `php -v` on the command prompt.

**Comment by FracturedSarcasm** [comment]
@Prophet731 Survey says:

![image](https://user-images.githubusercontent.com/49985936/80729993-db3ecf00-8abd-11ea-9d4b-ecaeac5ca060.png)

**Comment by FracturedSarcasm** [comment]
To add insult to injury, I started taking a closer look at the procon error logs and it looks like stat tracker is throwing an error similar to [error908](https://github.com/AdKats/AdKats/issues/908). My procon doesn't crash though. I have the latest version installed v1.0.3 and there were no issues setting it up. I did notice some duplicate player name entries being added to the DB. The main difference was the server ID and clan tag.

**Comment by Prophet731** [comment]
Does the BFACP have any logs in `storage/logs` from when you ran the command?

**Comment by FracturedSarcasm** [comment]
No. Only thing in that folder is the .gitignore file.

**Comment by Prophet731** [comment]
Copy and paste these commands in order.

```
php artisan optimize
php artisan cache:clear
php artisan config:clear
php artisan migrate --refresh
php artisan db:seed
`

**Comment by FracturedSarcasm** [comment]
![image](https://user-images.githubusercontent.com/49985936/80759944-f6750300-8aec-11ea-93af-f7fd8d41bb83.png)

Executed the commands in order. No logs. BFACP doesn't load.

**Comment by Prophet731** [comment]
Does anything happen if you do `php artisan list`

**Comment by FracturedSarcasm** [comment]
No. once I enter command, it goes back to command prompt.

**Comment by FracturedSarcasm** [comment]
I appreciate all the feedback dude. Honestly, I'll just go back to v2.0.3. I have my tweaked setup script and with your comment about the email issue, I can fix that too. I'd rather your attention be on getting v3 up and running. LOL.

I'll post an update once I make those changes, probably over the weekend. Then I'll close the issue once I have the dashboard up.

Again, thanks for feedback. It's been an awesome learning experience. I'm more of .NET w/Oracle dev and only dabbled in PHP years ago when work thought it would go that way for new projects.

**Comment by Prophet731** [comment]
Well if you're not getting anything from that command then the file isn't being run. Cause that command should have shown all the commands so something is up with your CLI PHP install.

As for V3, I haven't touched it in a couple of months now due to the COVID-19 thing and work stuff.

**Comment by FracturedSarcasm** [comment]
Tried a php reset on our hosted server and that apparently did the trick. I was able to run `php artisan list` and it did show a list of commands to execute. I still needed to force the setup scripts to run as they didn't do anything when the page first loaded. But with some tweaking, I was able to get v2.0.3 of the dashboard up and running (mostly). I had made changes to the .env.php file for the email settings and while it seemed to work, the user registration process still bombed. The error I get now was different from the error I got before I added the new email settings. So now I'm just poking around the site and trying to fix things that don't work, live scoreboard just sits and spins. Clicking on a player errors out, that's an issue with SQL so once I get back into the DB I can locate what the actual problem is.

As for V3, I understand. I'm just glad that it there's a prospect of it actually happening, even if it's a super tiny one. If nothing else, this has sparked my interest in learning about PHP.

Cheers!

**Comment by Prophet731** [comment]
I would run the laravel dev version I posted before. That has many fixes that 2.0.3 doesn't.

**Comment by FracturedSarcasm** [comment]
Sadly, that version did nothing for me. So I went back to v2.0.3 and I will just edit the code when shit breaks. I'm going to close the case, since I'm no longer have the original issue. Appreciate all the help. I'm keeping tabs on v3. haha
